src/transformers/models/bark/generation_configuration_bark.py

```diff
@@ -0,0 +1,258 @@
+        self.n_coarse_codebooks = n_coarse_codebooks
+        self.coarse_infer_token = coarse_infer_token
+        self.max_coarse_input_length = max_coarse_input_length
+        self.max_coarse_history = max_coarse_history
+        self.sliding_window_len = sliding_window_len
+
+        # 256
+
+
+class BarkFineGenerationConfig(GenerationConfig):
+    model_type = "fine_acoustics"
+
+    def __init__(
+        self,
+        temperature=0.5,
+        max_fine_history_length=512,
+        max_fine_input_length=1024,
+        n_fine_codebooks=8,
+        **kwargs,
+    ):
+        """_summary_
+
+        Args:
+            temperature (float, optional): _description_. Defaults to 0.5.
+            max_fine_history_length (int, optional): _description_. Defaults to 512.
+            max_fine_input_length (int, optional): _description_. Defaults to 1024.
+            n_fine_codebooks (int, optional): _description_. Defaults to 8.
+        """
+        super().__init__(temperature=temperature)
+
+        self.max_fine_history_length = max_fine_history_length
+        self.max_fine_input_length = max_fine_input_length
+        self.n_fine_codebooks = n_fine_codebooks
+
+
+class BarkGenerationConfig(GenerationConfig):
+    model_type = "bark"
+    is_composition = True
+
+    # TODO: nested from_dict
+
+    def __init__(
+        self,
+        semantic_config: Dict = None,
+        coarse_acoustics_config: Dict = None,
+        fine_acoustics_config: Dict = None,
+        sample_rate=24_000,
+        codebook_size=1024,
+        **kwargs,
+    ):
+        """_summary_
+
+        Args:
+            semantic_config (Dict, optional): _description_. Defaults to None.
+            coarse_acoustics_config (Dict, optional): _description_. Defaults to None.
+            fine_acoustics_config (Dict, optional): _description_. Defaults to None.
+            sample_rate (_type_, optional): _description_. Defaults to 24_000.
+            codebook_size (int, optional): _description_. Defaults to 1024.
+        """
+        if semantic_config is None:
+            semantic_config = {}
+            logger.info("semantic_config is None. initializing the semantic model with default values.")
+
+        if coarse_acoustics_config is None:
+            coarse_acoustics_config = {}
+            logger.info("coarse_acoustics_config is None. initializing the coarse model with default values.")
+
+        if fine_acoustics_config is None:
+            fine_acoustics_config = {}
+            logger.info("fine_acoustics_config is None. initializing the fine model with default values.")
+
+        self.semantic_config = BarkSemanticGenerationConfig(**semantic_config)
+        self.coarse_acoustics_config = BarkCoarseGenerationConfig(**coarse_acoustics_config)
+        self.fine_acoustics_config = BarkFineGenerationConfig(**fine_acoustics_config)
+
+        self.sample_rate = sample_rate
+        self.codebook_size = codebook_size
+
+    @classmethod
+    def from_configs(
```

Comment:
Perhaps a better API naming?

```suggestion
    def from_sub_model_configs(
```